extensions-core/kafka-extraction-namespace/src/test/java/io/druid/query/lookup/KafkaLookupExtractorFactoryTest.java

```diff
@@ -171,85 +164,60 @@ public void testCacheKeySameOnNoChange()
-        cacheManager,
-        TOPIC + "b",
-        DEFAULT_PROPERTIES
-    );
-    factory2.getMapRef().set(ImmutableMap.<String, String>of());
+    final KafkaLookupExtractorFactory factory1 = makeLookupFactory(cacheManager);
+    final KafkaLookupExtractorFactory factory2 = makeLookupFactory(cacheManager, TOPIC + 'b');
+
+    mockCacheManager();
+    factory1.init(); factory2.init();
 
     Assert.assertFalse(Arrays.equals(factory1.get().getCacheKey(), factory2.get().getCacheKey()));
   }
 
   @Test
   public void testReplaces()
   {
-    final KafkaLookupExtractorFactory factory = new KafkaLookupExtractorFactory(
-        cacheManager,
-        TOPIC,
-        DEFAULT_PROPERTIES
-    );
+    final KafkaLookupExtractorFactory factory = makeLookupFactory(cacheManager);
 
     Assert.assertTrue(factory.replaces(null));
 
     Assert.assertTrue(factory.replaces(new MapLookupExtractorFactory(ImmutableMap.<String, String>of(), false)));
     Assert.assertFalse(factory.replaces(factory));
-    Assert.assertFalse(factory.replaces(new KafkaLookupExtractorFactory(
-        cacheManager,
-        TOPIC,
-        DEFAULT_PROPERTIES
-    )));
+    Assert.assertFalse(factory.replaces(makeLookupFactory(cacheManager)));
 
-    Assert.assertTrue(factory.replaces(new KafkaLookupExtractorFactory(
-        cacheManager,
-        TOPIC + "b",
-        DEFAULT_PROPERTIES
-    )));
+    Assert.assertTrue(factory.replaces(makeLookupFactory(cacheManager, TOPIC + 'b')));
 
-    Assert.assertTrue(factory.replaces(new KafkaLookupExtractorFactory(
-        cacheManager,
-        TOPIC,
-        ImmutableMap.of("some.property", "some.other.value")
+    Assert.assertTrue(factory.replaces(makeLookupFactory(
+            cacheManager, ImmutableMap.of("some.property", "some.other.value")
     )));
 
-    Assert.assertTrue(factory.replaces(new KafkaLookupExtractorFactory(
-        cacheManager,
-        TOPIC,
-        ImmutableMap.of("some.other.property", "some.value")
+    Assert.assertTrue(factory.replaces(makeLookupFactory(
+        cacheManager, ImmutableMap.of("some.other.property", "some.value")
     )));
 
     Assert.assertTrue(factory.replaces(new KafkaLookupExtractorFactory(
         cacheManager,
         TOPIC,
         DEFAULT_PROPERTIES,
         1,
-        false
+        false,
+        NOOP_CONNECTION_FACTORY
     )));
 
     Assert.assertTrue(factory.replaces(new KafkaLookupExtractorFactory(
         cacheManager,
         TOPIC,
         DEFAULT_PROPERTIES,
         0,
-        true
+        true,
+        NOOP_CONNECTION_FACTORY
     )));
   }
 
   @Test
   public void testStopWithoutStart()
   {
-    final KafkaLookupExtractorFactory factory = new KafkaLookupExtractorFactory(
-        cacheManager,
-        TOPIC,
-        DEFAULT_PROPERTIES
-    );
-    Assert.assertTrue(factory.close());
+    final KafkaLookupExtractorFactory factory = makeLookupFactory(cacheManager);
+    Assert.assertFalse(factory.close());
```

Comment:
Why was this `true`? If the factory was never started, attempting to stop it should return `false`. Or am I missing something?